### DISCUSSION TITLE
Support Decimal passed to Client Components from Server Components

### DISCUSSION THREAD

**Discussion by omararturo** [vercel/next.js#68797] [Ideas]
### Goals

1. Allow server component send a Decimal object to client component

### Background

Nextjs support Date object. Decimal is a type used for many ORM, like Prisma. Manual conversion is a bad and unscalable solution.

### Proposal

I propose that Decimal was added to a communication from Server Component to Client Component like a Date, this feature will be good received by the community

**Comment by icyJoseph** [comment]
This sounds like something React has to implement, doesn't it?

- https://react.dev/reference/rsc/use-client#serializable-types

I'll reserve my opinions on trying to do this with an object that's custom defined in user land, and not a JavaScript native/primitive
